cli/dcoscli/package/main.py

```diff
@@ -206,6 +215,235 @@ def _remove_repo(repo_name):
     return 0
 
 
+def _build(build_definition,
+           output_directory):
+    """ Creates a DC/OS Package from a DC/OS Package Build Definition
+
+    :param build_definition: The path to a DC/OS Package Build Definition
+    :type build_definition: str
+    :param output_directory: The directory where the DC/OS Package
+    will be stored
+    :type output_directory: str
+    :returns: The process status
+    :rtype: int
+    """
+    # get the path of the build definition
+    cwd = os.getcwd()
+    build_definition_path = build_definition
+    if not os.path.isabs(build_definition_path):
+        build_definition_path = os.path.join(cwd, build_definition_path)
+
+    build_definition_directory = os.path.dirname(build_definition_path)
+
+    # get the path to the output directory
+    if output_directory is None:
+        output_directory = os.path.join(build_definition_directory, "target")
+
+        # ensure the directory exists
+        os.makedirs(output_directory, exist_ok=True)
+
+    logger.debug("Using [%s] as output directory", output_directory)
+
+    if not os.path.exists(build_definition_path):
+        raise DCOSException(
+            "The file [{}] does not exist".format(build_definition_path))
+
+    # load raw build definition
+    with util.open_file(build_definition_path) as bd:
+        build_definition_raw = util.load_json(bd, keep_order=True)
+
+    # validate DC/OS Package Build Definition with local references
+    build_definition_schema_path = "data/schemas/build-definition-schema.json"
+    build_definition_schema = util.load_jsons(
+        pkg_resources.resource_string(
+            "dcoscli", build_definition_schema_path).decode())
+
+    errs = util.validate_json(build_definition_raw, build_definition_schema)
+
+    if errs:
+        raise DCOSException('Error validating package: '
+                            '[{}] does not conform to the specified '
+                            'schema'.format(build_definition_path))
+
+    # resolve local references in build definition
+    _resolve_local_references(
+        build_definition_raw,
+        build_definition_schema,
+        build_definition_directory
+    )
+
+    # at this point all the local references have been resolved
+    build_definition_resolved = build_definition_raw
+
+    # validate resolved build definition
+    package_schema_path = "data/schemas/package-schema.json"
```

Comment:
Let's use `metadata-schema.json` instead. Would need to change all of the related variable names.